### DISCUSSION TITLE
[BUG] Video playback fails in apps (android and IOS) when adding custom proxy with self-signed SSL

### DISCUSSION THREAD

**Discussion by w00tlarr** [immich-app/immich#4757] [General]
### The bug

Initial Login to Immich app is OK where I'm able to set the https url and enabled the Allow self-signed SSL certificates 

Initial building of timeline and images are OK too and able to scroll all the way down the timeline with all thumbnails loaded properly.

Am able to view any static Images fine but I unable to playback any videos in the app itself when self-signed SSL.
Tho I am able to download the original video file.

Using external nginx as reverse proxy with this nginx.conf:

server {
    listen 2000 ssl default_server;
    listen [::]:2000 ssl default_server;
    server_name _;
  
    ssl_certificate /etc/certs/server.pem;
    ssl_certificate_key /etc/certs/server-key.pem;

    root /var/www/html;

    client_max_body_size 50000M;

    #"Immich"
    location / {
    proxy_pass_request_headers on;
        proxy_pass http://192.168.1.XX:2283;
        proxy_set_header Host              $http_host;
        proxy_set_header X-Real-IP         $remote_addr;
        proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_http_version 1.1;
        proxy_set_header   Upgrade    $http_upgrade;
        proxy_set_header   Connection "upgrade";
        proxy_redirect off;    
    }
  }

Log attached from app when playing videos:
[Immich_log_2023-10-26T15_10_30.207502.csv](https://github.com/immich-app/immich/files/13181765/Immich_log_2023-10-26T15_10_30.207502.csv)

### The OS that Immich Server is running on

Docker, managed with Portainer on Debian Bookworm. Using external nginx as reverse proxy.

### Version of Immich Server

1.82.1

### Version of Immich Mobile App

1.82.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Same as default in Github
```

### Your .env content

```Shell
Same as default in Github
```

### Reproduction steps

```bash
1. Import self-signed SSL (CA) into mobile device
2. Install Immich mobile app in Android/iOS
3. Set the Allow self-signed SSL certificates at the Login sceen with "gears" icon when setting up https url to nginx reverse proxy to Immich
4. Login to Immich
5. Allow initial timeline to be built
6. Noticed all static images are able to viewed normally
7. Start playing any video. Video is open and able to see the video thumbnail but is not able to play it. Loading indicator just spins and spins. Doesn't timeout or show any additional errors.
```

**Comment by w00tlarr** [comment]
nginx reverse proxy logs:
![image](https://github.com/immich-app/immich/assets/126688857/e723ee6b-2c40-41e1-90de-cd5e478c2eab)

**Comment by w00tlarr** [comment]
Discord: https://discord.com/channels/979116623879368755/1159225824935940167

**Comment by w00tlarr** [comment]
Forgot to mention - updated my nginx reverse proxy without SSL, setup Immich app with normal http url, and the video playback works OK

**Comment by t3dium** [comment]
I'm also encountering this issue, setup immich behind nginx-proxy-manager with a self signed cert and videos don't play on mobile. They only work on the web version of immich

**Comment by lwalegba** [comment]
I also had this issue. But changed to a certificate from Let's Encrypt (i.e. not self signed) and then it works. So it seems to be connected to just self signed certs.

**Reply by blacksong** [reply]
You are right

**Comment by w00tlarr** [comment]
> I'm also encountering this issue, setup immich behind nginx-proxy-manager with a self signed cert and videos don't play on mobile. They only work on the web version of immich

Yes. Issue only occurs at the mobile app

**Comment by blacksong** [comment]
I had the same issue using nginx and self-signed SSL.

**Comment by auzbaig94** [comment]
Same issue still with server version of 1.106.4 and app version on android (pixel 6a) 1.106.3.

Using self signed cert and nginx. Can stream in browser but not the app. Same with another iphone I have.

**Comment by jhagas** [comment]
I also encounter this issue, any update on this?

**Comment by w00tlarr** [comment]
refer to #5553
